integration/mod_test.go

```diff
@@ -1,157 +0,0 @@
-
-			response, err := http.Get(fmt.Sprintf("http://localhost:%s", container.HostPort()))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(response.StatusCode).To(Equal(http.StatusOK))
-
-			content, err := ioutil.ReadAll(response.Body)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(string(content)).To(ContainSubstring("go1.15"))
-
-			Expect(logs).To(ContainLines(
-				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, settings.Buildpack.Name)),
-				"  Executing build process",
-				fmt.Sprintf("    Running 'go build -o /layers/%s/targets/bin -buildmode pie .'", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
-				MatchRegexp(`      Completed in ([0-9]*(\.[0-9]*)?[a-z]+)+`),
-				"",
-				"  Assigning launch processes",
-				fmt.Sprintf("    web: /layers/%s/targets/bin/mod", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
-			))
-		})
-	})
-
-	context("when building an app that uses vendored modules in an offline environment", func() {
-		var (
-			image     occam.Image
-			container occam.Container
-
-			name   string
-			source string
-		)
-
-		it.Before(func() {
-			var err error
-			name, err = occam.RandomName()
-			Expect(err).NotTo(HaveOccurred())
-		})
-
-		it.After(func() {
-			Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
-			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
-			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
-			Expect(os.RemoveAll(source)).To(Succeed())
-		})
-
-		it("builds successfully", func() {
-			var err error
-			source, err = occam.Source(filepath.Join("testdata", "mod_vendor"))
-			Expect(err).NotTo(HaveOccurred())
-
-			var logs fmt.Stringer
-			image, logs, err = pack.Build.
-				WithPullPolicy("never").
-				WithNetwork("none").
-				WithBuildpacks(
-					settings.Buildpacks.GoDist.Offline,
-					settings.Buildpacks.GoBuild.Offline,
-				).
-				Execute(name, source)
-			Expect(err).ToNot(HaveOccurred(), logs.String)
-
-			container, err = docker.Container.Run.Execute(image.ID)
-			Expect(err).NotTo(HaveOccurred())
-
-			Eventually(container).Should(BeAvailable())
-
-			response, err := http.Get(fmt.Sprintf("http://localhost:%s", container.HostPort()))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(response.StatusCode).To(Equal(http.StatusOK))
-
-			content, err := ioutil.ReadAll(response.Body)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(string(content)).To(ContainSubstring("go1.14"))
-
-			Expect(logs).To(ContainLines(
-				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, settings.Buildpack.Name)),
-				"  Executing build process",
-				fmt.Sprintf("    Running 'go build -o /layers/%s/targets/bin -buildmode pie -mod vendor .'", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
-				MatchRegexp(`      Completed in ([0-9]*(\.[0-9]*)?[a-z]+)+`),
-				"",
-				"  Assigning launch processes",
-				fmt.Sprintf("    web: /layers/%s/targets/bin/mod_vendor", strings.ReplaceAll(settings.Buildpack.ID, "/", "_")),
-			))
-		})
-	})
-}
```

Comment:
I think that we actually probably still want this test but I don't know where it should go. Probably in a new test file.

Reply:
Actually, I'm having a hard time seeing the difference between the above test (in the case where the -mod vendor flag is used) and [this test](https://github.com/fg-j/go-build/blob/872fbc4c93b9daee6f0e19d27f51f8e0913b7dc3/integration/vendor_test.go#L95).

Reply:
The release notes linked in #66 state:

> When the main module contains a top-level vendor directory and its go.mod file specifies go 1.14 or higher, the go command now defaults to -mod=vendor for operations that accept that flag. A new value for that flag, -mod=mod, causes the go command to instead load modules from the module cache (as when no vendor directory is present).

Not clear from that what the expected behaviour is in the absence of a `go.mod`.

Reply:
We still have code that we want to test that relies on the presence of a `go.mod` file: https://github.com/paketo-buildpacks/go-build/blob/f9f97ca97cfc3ecbbe859ac4843bdf88c30e6c70/go_path_manager.go#L23-L26.

I think we can remove the vendored mod test and its fixture, but keep the non-vendored mod test.